### ISSUE TITLE
I want to verify the model performance in single frame mode（That is to say, remove the multi-frame fusion），What configuration needs to be changed?

### ISSUE THREAD

**Issue by EnginerPJ** [exiawsh/StreamPETR#60]
Is it enough to set memory_len, topk_proposals, and num_propagated to zero? Is there anything else that needs to be changed?
Looking forward to your reply！

**Comment by exiawsh** [comment]
> Is it enough to set memory_len, topk_proposals, and num_propagated to zero? Is there anything else that needs to be changed?
> Looking forward to your reply！

 Hi,
Here is the config of single frame model.
https://raw.githubusercontent.com/exiawsh/StreamPETR/main/projects/configs/PETRv1/focal_petrv1_r50_flash_704_24e.py

**Comment by EnginerPJ** [comment]
> > Is it enough to set memory_len, topk_proposals, and num_propagated to zero? Is there anything else that needs to be changed?
> > Looking forward to your reply！
> 
> Hi, Here is the config of single frame model. https://raw.githubusercontent.com/exiawsh/StreamPETR/main/projects/configs/PETRv1/focal_petrv1_r50_flash_704_24e.py

The accuracy of the model I trained in this way（set memory_len, topk_proposals, and num_propagated to zero） is better than that of the configuration file you mentioned（focal_petrv1_r50_flash_704_24e.py）
->set memory_len, topk_proposals, and num_propagated to zero：mAP=0.3674，NDS=0.4306
->focal_petrv1_r50_flash_704_24e.py：mAP=0.3644，NDS=0.4065

I'm wondering if my method is also a reasonable single-frame mode

**Comment by exiawsh** [comment]
@EnginerPJ Which config have you modified? Based on my experience, if you simply set those variables to zero. The results are always worse than the single frame model of my config.

**Comment by EnginerPJ** [comment]
> @EnginerPJ Which config have you modified? Based on my experience, if you simply set those variables to zero. The results are always worse than the single frame model of my config.

The configuration file I changed is this：projects/configs/StreamPETR/stream_petr_r50_flash_704_bs2_seq_428q_nui_60e.py
I directly set memory_len, topk_proposals, and num_propagated to zero, and did not make any changes to the others

**Comment by exiawsh** [comment]
@EnginerPJ The backbone of nui_60e has pretrained on nuimage, which is better than r50 in 1k pretraining. And you train the model 60 epochs. The config I provided only train the model 24 epochs.

**Comment by EnginerPJ** [comment]
> @EnginerPJ The backbone of nui_60e has pretrained on nuimage, which is better than r50 in 1k pretraining. And you train the model 60 epochs. The config I provided only train the model 24 epochs.
Yes, I noticed this, so I have changed the pre-trained model to be consistent with nui_60e

At the same time, nui_60e trains for 60 epochs, but the loss only converges to about 16. Although the one you provided only trains for 24 epochs, the loss can converge to about 14, so I am thinking of using the one you provided to train 60 epoch, can it further improve performance?

If I have time, I will submit another task to train the configuration file you provided to 60 epochs to see the effect.😊

**Comment by EnginerPJ** [comment]
> @EnginerPJ The backbone of nui_60e has pretrained on nuimage, which is better than r50 in 1k pretraining. And you train the model 60 epochs. The config I provided only train the model 24 epochs.

Yes, I noticed this, so I have changed the pre-trained model to be consistent with nui_60e

At the same time, nui_60e trains for 60 epochs, but the loss only converges to about 16. Although the one you provided only trains for 24 epochs, the loss can converge to about 14, so I am thinking of using the one you provided to train 60 epoch, can it further improve performance?

If I have time, I will submit another task to train the configuration file you provided to 60 epochs to see the effect.😊

**Comment by exiawsh** [comment]
@EnginerPJ How about the loss weight of velocity? The loss weight of single frame model should set to 0.2.

**Comment by EnginerPJ** [comment]
> velocity

Where can I set the weight of velocity loss? I can't find it in the config file

**Comment by EnginerPJ** [comment]
> @EnginerPJ How about the loss weight of velocity? The loss weight of single frame model should set to 0.2.

Where can I set the weight of velocity loss? I can't find it in the config file

**Comment by exiawsh** [comment]
@EnginerPJ code_weights = [2.0, 2.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 0.2, 0.2],

**Comment by EnginerPJ** [comment]
> @EnginerPJ code_weights = [2.0, 2.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 0.2, 0.2],

The weight corresponding to the single frame task I submitted in my own way is 1.0, I changed it to 0.2 and then resubmitted an experiment to see if the accuracy is further improved, thank you for reminding

**Comment by EnginerPJ** [comment]
My WeChat ID is 18256064073, can you add me as a friend, I want to discuss some technical issues with you, thank you very much
